src/__tests__/views/lottery/generateTicketNumbers.test.ts

```diff
@@ -0,0 +1,93 @@
+import generateTicketNumbers from 'views/LotteryV2/components/BuyTicketsModal/generateTicketNumbers'
+
+const withoutExistingNumbers = [
+  {
+    startingNumber: 1000000,
+    endingNumber: 1000099,
+    numbersToGenerate: 100,
+  },
+  {
+    startingNumber: 1998000,
+    endingNumber: 1999999,
+    numbersToGenerate: 2000,
+  },
+  {
+    startingNumber: 1156791,
+    endingNumber: 1161790,
+    numbersToGenerate: 5000,
+  },
+]
+
+const withExistingTickets = [
+  {
+    ...withoutExistingNumbers[0],
+    numbersToGenerate: 99,
```

Comment:
Interestingly the tests flagged that if a user were to buy... 999,901 tickets (at a cost of ~$5m in CAKE plus a massive amount in gas), and they tried to buy 100 more (exceeding the max ticket range by 1) - the function would crash. 

It's a fairly absurd edge case. I can write something to cater for this if we really think it necessary.